src/FrequencyMapAgent.hpp

```diff
@@ -64,12 +64,15 @@ namespace geopm
             enum m_policy_e {
                 M_POLICY_FREQ_DEFAULT,
                 M_POLICY_FREQ_UNCORE,
-		M_POLICY_FREQ_GPU_DEFAULT,
+                M_POLICY_FREQ_GPU_DEFAULT,
                 M_POLICY_FIRST_HASH,
                 M_POLICY_FIRST_FREQUENCY,
                 // The remainder of policy values can be additional pairs of
                 // (hash, frequency)
-                M_NUM_POLICY = 64,
+                // TODO: This number was 64. I changed it so the names vector
+                //       would be appropriately sized. Should I fix the logic
+                //       in the code instead?
+                M_NUM_POLICY = 65,
```

Comment:
Good call. Yeah, something like the following is one way it could be enforced as self-consistent:
```
M_POLICY_FREQ_DEFAULT,
M_POLICY_FREQ_UNCORE,
M_POLICY_FREQ_GPU_DEFAULT,
M_POLICY_AFTER_LAST_HEADER_FIELD,

M_POLICY_FIRST_HASH = M_POLICY_AFTER_LAST_HEADER_FIELD,
M_POLICY_FIRST_FREQUENCY,
M_POLICY_AFTER_LAST_RECURRING_FIELD,

// Note: 31 was originally chosen because it let us use a nice round total policy size of 64, but
// it's pretty arbitrary. Could probably just replace 31 with a named constant so readers know it is the
// number of regions we support in the FrequencyMapAgent's policy.
M_NUM_POLICY = 31 * (M_POLICY_AFTER_LAST_RECURRING_FIELD - M_POLICY_AFTER_LAST_HEADER_FIELD) + M_POLICY_AFTER_LAST_HEADER_FIELD
```

Reply:
If something like that is added, you could also prevent future similar issues if GPU freqs are added to the recurring part. Specifically, there are some places in the `.cpp` file that step through the recurring region of policies by `std::advance(it, 2)`. The `2` could instead use `M_POLICY_AFTER_LAST_RECURRING_FIELD - M_POLICY_AFTER_LAST_HEADER_FIELD` so it would stay consistent if a new type of frequency is added to the recurring part of the policy later on.